assembly/src/assembler/instruction/field_ops.rs

```diff
@@ -153,7 +153,14 @@ pub fn append_pow2_op(span_builder: &mut BasicBlockBuilder) {
 /// # Errors
 /// Returns an error if num_pow_bits is greater than 64.
 pub fn exp(span_builder: &mut BasicBlockBuilder, num_pow_bits: u8) -> Result<(), AssemblyError> {
-    validate_param(num_pow_bits, 0..=MAX_EXP_BITS)?;
+    if num_pow_bits > MAX_EXP_BITS {
+        return Err(AssemblyError::InvalidU8Param {
+            span: SourceSpan::UNKNOWN,
+            source_file: None,
```

Comment:
To fill these in properly, we should pass in the `ProcedureContext`, and populate those fields as in https://github.com/Himess/miden-vm/blob/89e4c8f9ba8d3c7ad034223220d2f1ffcecce094/assembly/src/assembler/mod.rs#L768-L769